util/chplenv/chpl_mem.py

```diff
@@ -8,18 +8,18 @@
 
 @memoize
 def get(flag='host'):
-    if flag == 'host':
-        mem_val = 'cstdlib'
-    elif flag == 'target':
-        mem_val = overrides.get('CHPL_MEM')
-        if not mem_val:
-            platform_val = chpl_platform.get('target')
-            cygwin = platform_val.startswith('cygwin')
-
-            if cygwin:
-                mem_val = 'cstdlib'
-            else:
-                mem_val = 'jemalloc'
+    platform_val = chpl_platform.get('target')
+    cygwin = platform_val.startswith('cygwin')
+    if flag == 'target':
+        if cygwin:
+            mem_val = 'cstdlib'
+        else:
+            mem_val = overrides.get('CHPL_TARGET_MEM') or overrides.get('CHPL_MEM') or 'cstdlib'
```

Comment:
I think this has changed the default target memory layer to be `cstdlib`

Reply:
The old style is more verbose and has more indentation, but I personally find that easier to follow compared to the `or` chaining

Reply:
Do you like this better?

```python
@memoize
def get(flag='host'):
    platform_val = chpl_platform.get('target')
    cygwin = platform_val.startswith('cygwin')
    chpl_host_mem = overrides.get('CHPL_HOST_MEM')
    chpl_target_mem = overrides.get('CHPL_TARGET_MEM')
    chpl_mem = overrides.get('CHPL_MEM')

    if flag == 'target':
        if cygwin:
            mem_val = 'cstdlib'
        elif chpl_target_mem:
            mem_val = chpl_target_mem
        elif chpl_mem:
            mem_val = chpl_mem
        else:
            mem_val = 'jemalloc'
    elif flag == 'host':
        if cygwin:
            mem_val = 'cstdlib'
        elif chpl_host_mem:
            mem_val = chpl_host_mem
        elif chpl_mem:
            mem_val = chpl_mem
        else:
            mem_val = 'cstdlib'
    else:
        error("Invalid flag: '{0}'".format(flag), ValueError)
    return mem_val
```

Reply:
Yeah, I like that. Definitely more code, but I find it easier to parse. Is the plan to eventually default to jemalloc for the host too, or do we imagine that being opt-in for now?

Reply:
I think opt-in for now until we get further testing done. Have updated to use the above